tools/mason/MasonPublish.chpl

```diff
@@ -0,0 +1,263 @@
+  }
+}
+
+/* Opens Spawn call to get username for the mason registry fork */
+private proc usernameCheck(username: string) {
+  const gitRemote = 'git ls-remote https://github/com';
+  var usernameCheck = (gitRemote + username + '/mason-registry').split();
+  var p = spawn(usernameCheck);
+  p.wait();
+  return p;
+}
+/* Runs Commands to see if Fork of mason-registry exists under the username */
+private proc checkIfForkExists(username: string) {
+  var getFork= ('git ls-remote https://github.com/' + username + '/mason-registry').split();
+  var p = spawn(getFork, stdout=PIPE);
+  p.wait();
+  return p;
+}
+
+/*Gets the GitHub username of the user, by parsing from the remote orgigin url.  */
+private proc getUsername() {
+  var usernameurl = gitUrl();
+  var tail = usernameurl.find("/")-1: int;
+  var head = usernameurl.find(":")+1: int;
+  var username = usernameurl(head..tail);
+  return username;
+}
+
+/* Clones the mason registery fork from the users repo. Takes username as input. */
+proc cloneMasonReg(username: string) throws {
+  try! {
+    const gitClone = 'git clone git@github.com:';
+    var ret = runCommand(gitClone  + username + "/mason-registry mason-registry", true);
+    return ret;
+  }
+  catch {
+    writeln('Error cloning the fork of mason-registry');
+    writeln('Make sure you have forked the mason-registry on GitHub');
+    exit(1);
+  }
+}
+
+/* Checks to see if 'git config --get remote.origin.url' exists */
+proc doesGitOriginExist() {
+  var urlExists = runCommand("git config --get remote.origin.url", true);
+  return !urlExists.isEmpty();
+}
+
+/*Procedure that reutns the url of the git remote origin */
+private proc gitUrl() {
+  var url = runCommand("git config --get remote.origin.url", true);
+  return url;
+}
+
+/* Takes the git username and creates a new branch of the mason registry users fork,
+ name or branch is taken from the Mason.toml of the mason package.*/
+proc branchMasonReg(username: string, name: string, safeDir: string) throws {
+  try! {
+    const masonReg = "/mason-registry/";
+    const branchCommand = "git checkout -b "+ name: string;
+    var ret = gitC(MASON_HOME + '/tmp/' + safeDir + '/mason-registry', branchCommand, true);
+    return ret;
+  }
+  catch {
+    here.chdir(MASON_HOME);
+    rmTree(safeDir);
+    writeln('Error branching mason-registry');
+    exit(1);
+  }
+}
+
+/* Gets name from the Mason.toml */
+private proc getPackageName() throws {
+  try! {
+    const cwd = getEnv("PWD");
+    const projectHome = getProjectHome(cwd);
+    const toParse = open(projectHome + "/Mason.toml", iomode.r);
+    var tomlFile = new owned(parseToml(toParse));
+    const name = tomlFile['brick']['name'].s;
+    return name;
+  }
+  catch {
+    here.chdir(MASON_HOME);
+    rmTree('/tmp');
+    writeln('Error getting the name of your package, ensure your package is a mason project');
+    exit(1);
+  }
+}
+
+/* Adds package to the Bricks of the mason-registry branch and then vrares the version.toml
```

Comment:
alias chplspell to spell, it'll save you alot of time and from these comments/iterations